lib/buffer.js

```diff
@@ -330,6 +331,33 @@ Buffer.from = function from(value, encodingOrOffset, length) {
   );
 };
 
+/**
+ * Creates the Buffer as a copy of the underlying ArrayBuffer of the TypedArray
+ * rather than the contents of the TypedArray.
+ * @param {ArrayBufferView} typedArray
+ * @param {number} [offset]
+ * @param {number} [length]
+ */
+Buffer.fromTypedArray = function fromTypedArray(typedArray, offset, length) {
+  if (!isArrayBufferView(typedArray)) {
+    throw new ERR_INVALID_ARG_TYPE('typedArray', 'TypedArray', typedArray);
+  }
+  offset ??= 0;
+  const typedArrayLength = TypedArrayPrototypeGetLength(typedArray);
+  length ??= typedArrayLength;
+
+  validateInteger(offset, 'offset', 0);
+  validateInteger(length, 'length', 0);
+
+  offset = MathMin(offset, typedArrayLength);
+  length = MathMin(length, typedArrayLength);
+
+  typedArray = TypedArrayPrototypeSlice(typedArray, offset, offset + length);
```

Comment:
I think we can skip the `TypedArrayPrototypeSlice` if we are copying the full `TypedArray`, right?
```suggestion
  if (offset != null && length != null) {
    offset ??= 0;
    const typedArrayLength = TypedArrayPrototypeGetLength(typedArray);
    length ??= typedArrayLength;

    validateInteger(offset, 'offset', 0);
    validateInteger(length, 'length', 0);

    offset = MathMin(offset, typedArrayLength);
    length = MathMin(length, typedArrayLength);

    typedArray = TypedArrayPrototypeSlice(typedArray, offset, offset + length);
  }
```